sklearn/preprocessing/data.py

```diff
@@ -1719,10 +1726,15 @@ def _transform_selected(X, transform, selected="all", copy=True):
         X_sel = transform(X[:, ind[sel]])
```

Comment:
Why is this not `X_sel = transform(X[:, sel])`? As long as `sel.shape[0] == n_features` (which it is), it can act as a valid slice.